src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1861,6 +1895,7 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
             if (leftResult.ValueCanBeNull() == false)
             {
                 ReportNonSafetyDiagnostic(ErrorCode.HDN_ExpressionIsProbablyNeverNull, leftOperand.Syntax);
+                SetUnreachable();
```

Comment:
>SetUnreachable [](start = 16, length = 14)

It looks like we're only using `SetUnreachable()` for cases where we can guarantee the null state. What are the implications of setting an unreachable state here where we cannot guarantee the state?

Reply:
I'm going to revert this change for now, and we'll talk about taking reachability into account in LDM at a later date.

---
In reply to: [246103019](https://github.com/dotnet/roslyn/pull/32225#discussion_r246103019) [](ancestors = 246103019,246099751)